lua/neogit/spinner.lua

```diff
@@ -0,0 +1,67 @@
+---@class Spinner
+---@field text string
+---@field count number
+---@field interval number
+---@field pattern string[]
+---@field timer uv_timer_t
+---@field start fun(self)
+---@field stop fun(self)
+local Spinner = {}
+Spinner.__index = Spinner
+
+---@return Spinner
+function Spinner.new(text)
+  local instance = {
+    text = text,
+    interval = 100,
+    count = 0,
+    timer = nil,
+    pattern = {
+      "⠋",
+      "⠙",
+      "⠹",
+      "⠸",
+      "⠼",
+      "⠴",
+      "⠦",
+      "⠧",
+      "⠇",
+      "⠏",
+    },
+  }
+
+  return setmetatable(instance, Spinner)
+end
+
+function Spinner:start()
+  if not self.timer then
+    self.timer = vim.uv.new_timer()
+    self.timer:start(
+      250,
+      self.interval,
+      vim.schedule_wrap(function()
+        self.count = self.count + 1
+        local step = self.pattern[(self.count % #self.pattern) + 1]
+        vim.cmd(string.format("redraw | echomsg '%s %s'", step, self.text))
```

Comment:
So `redraw | echomsg` is what allows the same notification to be updated?

I just tested with `noice.nvim` (I don't usually use it but it surfaces bad use of e.g. `vim.notify`) and it works nicely, I hadn't managed to do this myself:

![image](https://github.com/user-attachments/assets/d69053fd-5855-4337-82b1-eb4552331626)

Reply:
Yeah, `redraw` is commonly used to prevent hitting the "hit enter" prompt, which would otherwise happen. Very nice coincidence that it works with noice!